src/Microsoft.OData.ModelBuilder/Vocabulary/VocabularyTermConfigurationExtensions.cs

```diff
@@ -0,0 +1,289 @@
+using System.Linq;
+using Microsoft.OData.Edm;
+using Microsoft.OData.Edm.Vocabularies;
+
+using Microsoft.OData.ModelBuilder.Capabilities.V1;
+using Microsoft.OData.ModelBuilder.Core.V1;
+
+namespace Microsoft.OData.ModelBuilder
+{
+	/// <summary>
+	/// Extension methods for vocabulary builders configurations
+	/// </summary>
+	public static class VocabularyTermConfigurationExtensions
+	{
+		internal static IEdmExpression ToEdmExpression(this string text)
+			=> string.IsNullOrEmpty(text) ? null: new EdmStringConstant(text);
+		/// <summary>
+		/// <see cref="CallbackSupportedConfiguration"/> configuration
+		/// </summary>
+		/// <typeparam name="TEntity">The entity type of the navigation source.</typeparam>
+		/// <param name="navigationSource">The <see cref="IEdmNavigationSource"/> that can be built using <see cref="ODataModelBuilder"/>.</param>
+		/// <returns><see cref="CallbackSupportedConfiguration"/></returns>
+		public static CallbackSupportedConfiguration HasCallbackSupported<TEntity>(this EntitySetConfiguration<TEntity> navigationSource) where TEntity : class
+			=> navigationSource?.Configuration.VocabularyTermConfigurations.GetOrCreateConfiguration<CallbackSupportedConfiguration>();
+
+		/// <summary>
+		/// <see cref="ChangeTrackingConfiguration"/> configuration
+		/// </summary>
+		/// <typeparam name="TEntity">The entity type of the navigation source.</typeparam>
+		/// <param name="navigationSource">The <see cref="IEdmNavigationSource"/> that can be built using <see cref="ODataModelBuilder"/>.</param>
+		/// <returns><see cref="ChangeTrackingConfiguration"/></returns>
+		public static ChangeTrackingConfiguration HasChangeTracking<TEntity>(this NavigationSourceConfiguration<TEntity> navigationSource) where TEntity : class
+			=> navigationSource?.Configuration.VocabularyTermConfigurations.GetOrCreateConfiguration<ChangeTrackingConfiguration>();
+
+		/// <summary>
+		/// <see cref="ChangeTrackingConfiguration"/> configuration
+		/// </summary>
+		/// <param name="operationSource">The <see cref="IEdmOperation"/> that can be built using <see cref="ODataModelBuilder"/>.</param>
+		/// <returns><see cref="ChangeTrackingConfiguration"/></returns>
+		public static ChangeTrackingConfiguration HasChangeTracking(this FunctionConfiguration operationSource)
+			=> operationSource?.VocabularyTermConfigurations.GetOrCreateConfiguration<ChangeTrackingConfiguration>();
+
+		/// <summary>
+		/// <see cref="CountRestrictionsConfiguration"/> configuration
+		/// </summary>
+		/// <typeparam name="TEntity">The entity type of the navigation source.</typeparam>
+		/// <param name="navigationSource">The <see cref="IEdmNavigationSource"/> that can be built using <see cref="ODataModelBuilder"/>.</param>
+		/// <returns><see cref="CountRestrictionsConfiguration"/></returns>
+		public static CountRestrictionsConfiguration HasCountRestrictions<TEntity>(this EntitySetConfiguration<TEntity> navigationSource) where TEntity : class
```

Comment:
> EntitySetConfiguration [](start = 81, length = 23)

where's "SingletonConfiguration<TEntity>" ?

Reply:
Soo...
I am using the capabilities vocabulary model that is added to a default EdmModel...
```var model = new EdmModel().ReferencedModels.FirstOrDefault(m => m.GetNamespaceAlias("Org.OData.Capabilities.V1") != null);```

This model, which has been embedded in the Microsoft.OData.Edm lib has the following definition for CountRestrictions
```xml
[<Term Name="CountRestrictions" Type="Capabilities.CountRestrictionsType" AppliesTo="EntitySet">](https://github.com/OData/odata.net/blob/f375e468b77c4c16445e6e15c0b9afcbe54d387b/src/Microsoft.OData.Edm/Vocabularies/CapabilitiesVocabularies.xml#L240)
```
Which is different from the oasis specification 
```xml
[<Term Name="CountRestrictions" Type="Capabilities.CountRestrictionsType" AppliesTo="EntitySet Singleton">](https://github.com/oasis-tcs/odata-vocabularies/blob/604e8c41116e6ce343fbd91e3ced4153bb40dcd7/vocabularies/Org.OData.Capabilities.V1.xml#L221)
```

Created a [new issue in OData.net](https://github.com/OData/odata.net/issues/1835)